src/Umbraco.Web.UI.Client/src/libs/observable-api/states/array-state.ts

```diff
@@ -61,6 +63,27 @@ export class UmbArrayState<T> extends UmbDeepState<T[]> {
 		}
 	}
 
+	/**
+	 * @function hasOne
+	 * @param {U} unique - the unique value to compare with.
+	 * @returns {boolean} Wether it existed
+	 * @description - Check if a unique value exists in the current data of this Subject.
+	 * @example <caption>Example check for key to exist.</caption>
+	 * const data = [
+	 * 	{ key: 1, value: 'foo'},
+	 * 	{ key: 2, value: 'bar'}
+	 * ];
+	 * const myState = new UmbArrayState(data, (x) => x.key);
+	 * myState.hasOne(1);
+	 */
+	hasOne(unique: U): boolean {
```

Comment:
I believe the correct name of this method should be "getHasOne" to follow our naming convention. I would expect this to return an observable 🤔

Reply:
Oh yes, good one

Reply:
I have pushed an update